packages/components/src/draggable/index.js

```diff
@@ -156,30 +208,37 @@ export default function Draggable( {
 		}
 
 		// Mark the current cursor coordinates.
-		let cursorLeft = event.clientX;
-		let cursorTop = event.clientY;
+		const dragState = {
+			x,
+			y,
+			cursorLeft: event.clientX,
+			cursorTop: event.clientY,
+			cloneWrapper,
+			scale,
+		};
 
 		/**
-		 * @param {import('react').DragEvent} e
+		 * @param {import("react").DragEvent<Element>} e
 		 */
-		function over( e ) {
-			cloneWrapper.style.top = `${
-				parseInt( cloneWrapper.style.top, 10 ) + e.clientY - cursorTop
-			}px`;
-			cloneWrapper.style.left = `${
-				parseInt( cloneWrapper.style.left, 10 ) + e.clientX - cursorLeft
-			}px`;
-
-			// Update cursor coordinates.
-			cursorLeft = e.clientX;
-			cursorTop = e.clientY;
-
-			if ( onDragOver ) {
-				onDragOver( e );
+		function update( e ) {
+			//Skip doing any work if mouse has not moved.
+			if (
+				dragState.cursorLeft === e.clientX &&
+				dragState.cursorTop === e.clientY
+			) {
+				return;
 			}
+			over( e, dragState );
 		}
 
-		ownerDocument.addEventListener( 'dragover', over );
+		// Aim for 60fps (16 ms per frame) for now. We can potentially use requestAnimationFrame (raf) instead,
+		// note that browsers may throttle raf below 60fps in certain conditions.
+		const throttledDragOver = throttle( update, 16 );
+
+		/**
+		 * @param {import('react').DragEvent} e
+		 */
+		ownerDocument.addEventListener( 'dragover', throttledDragOver );
```

Comment:
I think `throttledDragOver` will have to be removed as the event listener on line 270.

Reply:
Great catch! ✨ Updated in https://github.com/WordPress/gutenberg/pull/33395/commits/4f4294366dda0275ea124e907eb5634d3291af08